.github/workflows/build.yml

```diff
@@ -10,18 +10,27 @@ on:
 
 jobs:
     windows:
-        runs-on: windows-latest
+        runs-on: ${{matrix.os}}
         name: "Windows: Build and test"
         defaults:
             run:
                 shell: cmd
         strategy:
             fail-fast: false
             matrix:
-                php: ["7.3", "7.4", "8.0", "8.1", "8.2", "8.3"]
+                os: [ windows-2019, windows-2022 ]
+                php: ["7.3", "7.4", "8.0", "8.1", "8.2", "8.3","8.4"]
                 arch: [x86, x64]
                 ts: [nts, ts]
                 experimental: [false]
+                exclude:
+                  - { os: windows-2019, php: "8.4" }
+                  - { os: windows-2019, php: "8.3" }
+                  - { os: windows-2019, php: "8.2" }
+                  - { os: windows-2019, php: "8.1" }
+                  - { os: windows-2019, php: "8.0" }
+                  - { os: windows-2022, php: "7.4" }
+                  - { os: windows-2022, php: "7.3" }
```

Comment:
Wouldn't it be better to just spell out all the *right* combinations instead of the *wrong* ones?